modconfig/flowpipe_credential.go

```diff
@@ -1880,6 +1880,80 @@ func (c *BitbucketCredential) Validate() hcl.Diagnostics {
 	return hcl.Diagnostics{}
 }
 
+type DatadogCredential struct {
+	HclResourceImpl
+	ResourceWithMetadataImpl
+
+	Type string `json:"type" cty:"type" hcl:"type,label"`
+
+	APIKey *string `json:"api_key,omitempty" cty:"api_key" hcl:"api_key,optional"`
+	AppKey *string `json:"app_key,omitempty" cty:"app_key" hcl:"app_key,optional"`
+	APIUrl *string `json:"api_url,omitempty" cty:"api_url" hcl:"api_url,optional"`
+}
+
+func (*DatadogCredential) GetCredentialType() string {
+	return "datadog"
+}
+
+func (c *DatadogCredential) getEnv() map[string]cty.Value {
+	env := map[string]cty.Value{}
+	if c.APIKey != nil {
+		env["DD_CLIENT_API_KEY"] = cty.StringVal(*c.APIKey)
+	}
+	if c.AppKey != nil {
+		env["DD_CLIENT_APP_KEY"] = cty.StringVal(*c.AppKey)
+	}
+	return env
+}
+
+func (c *DatadogCredential) CtyValue() (cty.Value, error) {
+	ctyValue, err := GetCtyValue(c)
+	if err != nil {
+		return cty.NilVal, err
+	}
+
+	valueMap := ctyValue.AsValueMap()
+	valueMap["env"] = cty.ObjectVal(c.getEnv())
+
+	return cty.ObjectVal(valueMap), nil
+}
+
+func (c *DatadogCredential) Resolve(ctx context.Context) (Credential, error) {
+	datadogAPIKeyEnvVar := os.Getenv("DD_CLIENT_API_KEY")
+	datadogAppKeyEnvVar := os.Getenv("DD_CLIENT_APP_KEY")
+
+	// Don't modify existing credential, resolve to a new one
+	newCreds := &DatadogCredential{
+		HclResourceImpl: HclResourceImpl{
+			FullName:        c.FullName,
+			UnqualifiedName: c.UnqualifiedName,
+			ShortName:       c.ShortName,
+			DeclRange:       c.DeclRange,
+			blockType:       c.blockType,
+		},
+		Type:   c.Type,
+		APIUrl: c.APIUrl,
+	}
+
+	if c.APIKey == nil {
```

Comment:
This is a bug ... what if c.APIKey is NOT null?

Reply:
ah, added the default in that case